include/dump_offload.hpp

```diff
@@ -341,76 +341,100 @@ inline void resetHandlers()
     }
 }
 
+inline void dumpOffloadStreamingOnOpenCallback(
+    crow::streaming_response::Connection& conn)
+{
+    if (!systemHandlers.empty())
+    {
+        BMCWEB_LOG_WARNING("Can't allow dump offload opertaion, one "
+                           "of the host dump is already offloading");
+        conn.sendStreamErrorStatus(
+            boost::beast::http::status::service_unavailable);
+        conn.close();
+        return;
+    }
+
+    std::string url(conn.req.target());
+    std::string startDelimiter = "Entries/";
+    std::size_t pos1 = url.rfind(startDelimiter);
+    std::size_t pos2 = url.rfind("/attachment");
+    if (pos1 == std::string::npos || pos2 == std::string::npos)
+    {
+        BMCWEB_LOG_WARNING("Unable to extract the dump id");
+        conn.sendStreamErrorStatus(boost::beast::http::status::not_found);
+        conn.close();
+        return;
+    }
+    std::string dumpId = url.substr(pos1 + startDelimiter.length(),
+                                    pos2 - pos1 - startDelimiter.length());
+    std::string dumpType = "system";
+
+    boost::asio::io_context* ioCon = conn.getIoContext();
+
+    // Generating random id to create unique socket file
+    // for each dump offload request
+    std::random_device rd;
+    std::default_random_engine gen(rd());
+    std::uniform_int_distribution<> dist{0, 1024};
+    std::string unixSocketPath = std::string(unixSocketPathDir) + dumpType +
+                                 "_dump_" + std::to_string(dist(gen));
+    systemHandlers[&conn] = std::make_shared<Handler>(*ioCon, dumpId, dumpType,
+                                                      unixSocketPath);
+    systemHandlers[&conn]->connection = &conn;
+
+    if (!crow::ibm_utils::createDirectory(unixSocketPathDir))
+    {
+        systemHandlers[&conn]->connection->sendStreamErrorStatus(
+            boost::beast::http::status::not_found);
+        systemHandlers[&conn]->connection->close();
```

Comment:
what happens to object created at line 381? Since we are using the connection address as the key to the map there is not guarantee that the object will be reinitialised at the same map location, hence there is a possibility of memory leak.